src/mon/OSDMonitor.cc

```diff
@@ -74,18 +74,18 @@
 #define OSD_PG_CREATING_PREFIX "osd_pg_creating"
 
 struct C_UpdateCreatingPGs : public Context {
-  OSDMonitor *mon;
+  OSDMonitor *osdmon;
   utime_t start;
   epoch_t epoch;
-  C_UpdateCreatingPGs(OSDMonitor *mon, epoch_t e) :
-    mon(mon), start(ceph_clock_now()), epoch(e) {}
+  C_UpdateCreatingPGs(OSDMonitor *osdmon, epoch_t e) :
+    osdmon(osdmon), start(ceph_clock_now()), epoch(e) {}
   void finish(int r) override {
     if (r >= 0) {
       utime_t end = ceph_clock_now();
       dout(10) << "osdmap epoch " << epoch << " mapping took "
 	       << (end - start) << " seconds" << dendl;
-      mon->update_creating_pgs();
-      mon->check_pg_creates_subs();
+      osdmon->update_creating_pgs();
+      osdmon->check_pg_creates_subs();
```

Comment:
It looks to me like all of these locations already hold Monitor::lock... I'm not sure this is the problem?